libraries/rush-lib/src/api/CustomTips.ts

```diff
@@ -0,0 +1,88 @@
+import { JsonFile } from '@rushstack/node-core-library';
+
+export interface IRushCustomTipsJson {
+  prefix?: string;
+  customTips?: CustomTipItem[];
+}
+
+export type TipLevel = 'error' | 'warn' | 'info';
+export interface CustomTipItem {
+  id: CustomTipId;
+  prefix?: string;
+  level: TipLevel;
+  tip: '';
+}
+
+// TODO: consider making this work with the plugin (e.g., the plugins are able to define their own customizable tips)
+export type CustomTipId = 'PNPM_MISMATCH_DEPENDENCY' | 'ANOTHER_ERROR_ID';
+
+/**
+ * Singleton. Should always be used with getInstance(). This is to avoid loading the config file (`rush-custom-tips.json`) multiple times.
```

Comment:
Would favor an accessor off of the current `RushConfiguration` instance that caches the `CustomTips` (or probably `CustomTipsConfiguration` to match the convention for other config file abstractions).

See https://github.com/microsoft/rushstack/blob/main/libraries/rush-lib/src/api/ExperimentsConfiguration.ts for an object with similar usage.

We don't like generally like singletons in Rush for various reasons:
1) If a tool modifies the config file and needs to reinitialize, how do you do it?
2) The library can and will be used at times to operate on multiple distinct Rush repositories in the same NodeJS process.
3) They're awkward to unit test

Reply:
> Would favor an accessor off of the current RushConfiguration instance that caches the CustomTips

I see. I also considered that but I thought that would require everywhere to receive this instance (e.g., the `VersionMismatchFinder.ensureConsistentVersions` method need to receive another option (in addition to current `variant`)). But that might be better than the singleton. Will change that.

Reply:
changed accordingly